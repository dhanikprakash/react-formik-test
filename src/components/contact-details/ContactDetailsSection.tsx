import React, { Suspense, useState } from "react";

import { Spinner } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import { ContactDetailsBO } from "../../../types/contact-details";
import { Field, Form, Formik } from "formik";
import { validationSchema } from "./ContactDetailsValidation";
import Button from "@mui/material/Button";
import { toTitleCase } from "../utils/to-title-case";

export const ContactDetailsSection: React.FC<{
  contactDetails: ContactDetailsBO;
}> = ({ contactDetails: contactDetailsProps }) => {
  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
  const { emailAddress, mobileNumber, homePhoneNumber, workPhoneNumber } =
    contactDetailsProps;

  const { t } = useTranslation("contact-details");

  const onChangeSuccess = (): void => {};

  return (
    <>
      <Suspense fallback={<Spinner animation="border" />}>
        <h3>Contact Details</h3>
        <Formik
          initialValues={{
            emailAddress: emailAddress ?? "",
            mobileNumber: mobileNumber ?? "",
            homePhoneNumber: homePhoneNumber ?? "",
            workPhoneNumber: workPhoneNumber ?? "",
          }}
          validationSchema={validationSchema(t)}
          onSubmit={({
            emailAddress,
            mobileNumber,
            homePhoneNumber,
            workPhoneNumber,
          }): void => {
            setIsSubmitting(true);

            console.log(
              `submitting : ${emailAddress} ${mobileNumber} ${homePhoneNumber} ${workPhoneNumber} ${isSubmitting}`
            );
            //api-call to update data
            //success
            setIsSubmitting(false);
            onChangeSuccess();
            //failure
            //setIsSubmitting(false);
          }}
        >
          {(formik): React.ReactNode => (
            <Form>
              <div>
                {toTitleCase(t("emailAddress.label"))}
                <Field
                  type="text"
                  name="emailAddress"
                  title={t("emailAddress.tooltip")}
                  placeholder="Email"
                ></Field>
              </div>
              <div>
                {t("mobileNumber.label")}
                <Field
                  type="text"
                  name="mobileNumber"
                  title={t("mobileNumber.tooltip")}
                  placeholder="MobileNumber"
                ></Field>
              </div>
              <div>
                {t("homePhoneNumber.label")}
                <Field
                  type="text"
                  name="homePhoneNumber"
                  title={t("homePhoneNumber.tooltip")}
                  placeholder="Home Phone Number"
                ></Field>
              </div>
              <div>
                {t("workPhoneNumber.label")}
                <Field
                  type="text"
                  name="workPhoneNumber"
                  title={t("workPhoneNumber.tooltip")}
                  placeholder="workPhoneNumber"
                ></Field>
              </div>
              <div>
                <br />
                <Button variant="contained" onClick={formik.submitForm}>
                  Update
                </Button>
              </div>
            </Form>
          )}
        </Formik>
      </Suspense>
    </>
  );
};
