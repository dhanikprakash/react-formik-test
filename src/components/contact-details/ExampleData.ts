import { ContactDetailsBO } from "../../../types/contact-details";


export const exampleContactDetails: ContactDetailsBO = {
   emailAddress: "test@test.com", 
   mobileNumber: "1234567891",
   homePhoneNumber: "1234567891",
   workPhoneNumber: "1234567891"
}