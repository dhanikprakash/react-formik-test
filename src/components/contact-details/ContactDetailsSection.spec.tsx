import React from "react";
import * as renderer from "react-test-renderer";
import { ContactDetailsSection } from "./ContactDetailsSection";
import { exampleContactDetails } from "./ExampleData";

import i18n from "i18next";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  resources: {
    en: {},
  },
  lng: "en",
  fallbackLng: "en",
  interpolation: {
    escapeValue: false, // not needed for react as it escapes by default
  },
});

test("ContactDetailsSection should render correctly", () => {
  const data = renderer
    .create(<ContactDetailsSection contactDetails={exampleContactDetails} />)
    .toJSON();

  expect(data).toMatchSnapshot();
});
