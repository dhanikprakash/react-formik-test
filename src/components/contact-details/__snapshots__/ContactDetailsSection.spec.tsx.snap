// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetailsSection should render correctly 1`] = `
Array [
  <h3>
    Contact Details
  </h3>,
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div>
      Emailaddress.label
      <input
        name="emailAddress"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
        title="emailAddress.tooltip"
        type="text"
        value="test@test.com"
      />
    </div>
    <div>
      mobileNumber.label
      <input
        name="mobileNumber"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="MobileNumber"
        title="mobileNumber.tooltip"
        type="text"
        value="1234567891"
      />
    </div>
    <div>
      homePhoneNumber.label
      <input
        name="homePhoneNumber"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Home Phone Number"
        title="homePhoneNumber.tooltip"
        type="text"
        value="1234567891"
      />
    </div>
    <div>
      workPhoneNumber.label
      <input
        name="workPhoneNumber"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="workPhoneNumber"
        title="workPhoneNumber.tooltip"
        type="text"
        value="1234567891"
      />
    </div>
    <div>
      <br />
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Update
      </button>
    </div>
  </form>,
]
`;
